src/main/java/appeng/container/implementations/ContainerMEMonitorable.java

```diff
@@ -122,9 +123,22 @@ else if( monitorable instanceof IMEChest )
 				{
 					this.setPowerSource( (IEnergySource) monitorable );
 				}
-				else if( monitorable instanceof IGridHost )
+				else if( monitorable instanceof IGridHost || monitorable instanceof IActionHost )
 				{
-					final IGridNode node = ( (IGridHost) monitorable ).getGridNode( AEPartLocation.INTERNAL );
+					final IGridNode node;
+					if( monitorable instanceof IGridHost )
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)